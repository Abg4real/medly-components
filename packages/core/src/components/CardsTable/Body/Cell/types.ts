import { ColumnConfig } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
