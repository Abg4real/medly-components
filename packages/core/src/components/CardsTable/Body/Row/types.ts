import { WithThemeProp } from '@medly-components/utils';

export interface Props extends WithThemeProp {
    onClick?: () => void;
    withWhiteBackground: boolean;
}
