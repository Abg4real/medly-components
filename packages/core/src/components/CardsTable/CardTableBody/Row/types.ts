import { Props as CardProps } from '../../../Card/types';

export interface Props extends CardProps {
    disabled?: boolean;
    withLightTheme: boolean;
}
