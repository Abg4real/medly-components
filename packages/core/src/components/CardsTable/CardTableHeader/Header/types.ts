import { SortOrder } from '../../types';
import { WithThemeProp } from '@medly-components/utils/src';

export interface Props extends WithThemeProp{
    sortField?: string;
    field?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
    enableSorting?: boolean;
    title: string;
}    