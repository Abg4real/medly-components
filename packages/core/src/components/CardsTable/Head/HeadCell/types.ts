import { WithThemeProp } from '@medly-components/utils/src';
import { ColumnConfig, SortOrder } from '../../types';

export interface Props extends WithThemeProp {
    column: ColumnConfig;
    sortField?: string;
    onSortChange?: (field: string, order: SortOrder) => void;
}

export interface StyledProps {
    flex: number;
    align: 'left' | 'center' | 'right';
}
