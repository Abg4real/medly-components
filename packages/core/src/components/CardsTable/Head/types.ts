import { ColumnConfig, SortOrder } from '../types';

export interface Props {
    columns: ColumnConfig[];
    onSort?: (field: string, order: SortOrder) => void;
}
