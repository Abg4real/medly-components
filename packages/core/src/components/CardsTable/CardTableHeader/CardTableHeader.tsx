import React, { SFC, useCallback, useState } from "react";
import Card from "../../Card";
import { Props } from "./types";
import { SortOrder } from "../types";
import Header from "./Header";


export const CardTableHeader: SFC<Props> = React.memo( props => {

    const { onSort, columns } = props;
    const [sortField, setSortField] = useState(''),
        handleSortChange = useCallback(
            (field: string, order: SortOrder) => {
                setSortField(field);
                onSort(field, order);
            },
            [onSort]
        );
    
    return (
        <Card flowDirection="horizontal" variant="flat" fullWidth >
            {columns.map((column, index) => {
                return (
                    <Header 
                        key={index}
                        sortField={sortField}
                        field={column.field}
                        onSortChange={handleSortChange}
                        enableSorting={column.sort}
                        title={column.title}
                    />
                )
            })}
        </Card>
    )   


});

CardTableHeader.displayName = 'CardTableHeader';
