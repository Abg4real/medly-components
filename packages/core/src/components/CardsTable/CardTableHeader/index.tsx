import { CardTableHeader } from './CardTableHeader';

export default CardTableHeader;
