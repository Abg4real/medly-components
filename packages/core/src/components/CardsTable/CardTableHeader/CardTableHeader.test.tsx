import { render } from '@test-utils';
import React from 'react';
import { CardTableHeader } from './CardTableHeader'
import { ColumnConfig } from '../types';

describe('CardTableHeader', () => {
    it('should render CardTableHeader of cards table properly', () => {

        const onSort = jest.fn(),
        columns: ColumnConfig[] = [
            { title: 'Patient Info', field: 'patientInfo', formatter: 'text-short', sort: true },
            { title: 'RxInfo', field: 'rxInfo', formatter: 'group', sort: true }
        ];
        const { container } = render(<CardTableHeader 
            onSort={onSort}
            columns={columns}

         />);
        expect(container).toMatchSnapshot();
    });
});
