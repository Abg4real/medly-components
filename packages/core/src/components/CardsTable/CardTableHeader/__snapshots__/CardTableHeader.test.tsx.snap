// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTableHeader should render CardTableHeader of cards table properly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  padding: 10px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  width: calc(100% - 10px);
}

.c0.c0 * {
  background-color: transparent;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
  width: calc(100% - 10px);
}

.c1.c1 * {
  background-color: transparent;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  margin: 5px;
  overflow: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  box-sizing: border-box;
  background-color: #ffffff;
  padding: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  border-radius: 4px;
  border: 1px solid #dfe4e9;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 0;
  margin: 0;
  border: none;
  box-shadow: none;
  background-color: transparent;
}

.c3.c3 * {
  background-color: transparent;
}

.c6 {
  font-size: 1.6rem;
  cursor: pointer;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
}

.c6 * {
  fill-opacity: 1;
  fill: #607890;
}

.c5 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
}

.c2 {
  padding: 0 10px;
}

.c2 .c4 {
  color: #435465;
  text-transform: uppercase;
  font-size: 1.2rem;
  -webkit-letter-spacing: 0.2rem;
  -moz-letter-spacing: 0.2rem;
  -ms-letter-spacing: 0.2rem;
  letter-spacing: 0.2rem;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .sc-AykKF {
  cursor: pointer;
  margin-right: 1.0rem;
  padding-left: 1.0rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 c2"
      title="Patient Info"
    >
      <div
        class="c3"
      >
        <span
          class="c4 c5"
        >
          Patient Info
          <svg
            class="c6"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      class="c1 c2"
      title="RxInfo"
    >
      <div
        class="c3"
      >
        <span
          class="c4 c5"
        >
          RxInfo
          <svg
            class="c6"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.208 5.03a1 1 0 0 1 1.584 0l3.165 4.11a1 1 0 0 1-.793 1.61H8.836a1 1 0 0 1-.793-1.61l3.165-4.11zM12.792 18.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
              fill="#000"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
